### ISSUE TITLE
Why my settings for CustomRightClickMenu are lost in the plugin several times?

### ISSUE THREAD

**Issue by pendave** [SanderRonde/CustomRightClickMenu#29]
Hi, it happened several times. I don't know how the settings become blank. So I have to import the backed text file for the settings.

Is it a bug?

Could you do something backing the settings to online services such as  Google Drive, Dropbox, Onedrive ... which are existed in Tampermonkey toolkit.
![tw hr a289_lu8hloqechkg](https://user-images.githubusercontent.com/1331881/50441223-a6a08d80-0934-11e9-9c2d-8f1c5a23889c.png)

**Comment by SanderRonde** [comment]
The extension does "backup" your settings somewhat. The settings are uploaded to chrome.storage.sync, which stores your settings on chrome's servers. Restoring from backup would also not work as an issue with the settings is (most likely) the reason that the options page fails to load. Restoring from backup would then just trigger the same error. 

Regarding the white screen, it does sound like a bug but I haven't seen it yet. Could you post a screenshot of the dev tools when it happens? That allows me to figure out what's causing it. To do that press F12, go to the console tab and make a screenshot of the page.

**Comment by pendave** [comment]
![2019-01-10_220251](https://user-images.githubusercontent.com/1331881/50973360-ac48c680-1523-11e9-8ad7-2cad303a8feb.jpg)

It happens again.

**Comment by SanderRonde** [comment]
Unfortunately I can't do a lot to fix this bug without any information about when and how it occurred. I also had a look into the code again to find any possible functions that clear all settings and found that there is only a single function that clears your settings and that's the one that runs on the initial installation.

I think the most likely reason for your data being lost is you accidentally deleting your chrome storage. Extensions, programs or buttons that claim to "remove junk" generally remove cookies and the like, which would include removing your settings for this extension. Check if the settings for any other extensions are reset at the same time that they are for this extension. If they are, that would mean that all settings files are removed and if they are not, the problem would be more likely to be in this extension.

**Comment by BrockA** [comment]
I was having this problem too.  Menu resets to "factory" when Firefox restarts.  
Disabling the "_Sync your storage across all browser instances_" option _seems_ to fix it so far.

**Comment by SanderRonde** [comment]
I do think I've found the reason for the original issue happening. When you log into your browser on another computer and launch this extension for the first time on that computer, your settings are overwritten globally. This then resets all your settings to the default settings. However, this only happens if you log into another browser instance. If it happens when you just restart firefox, I think the issue may be with your sync settings not being saved in general, which might indicate an issue with your account or installation.

**Comment by BrockA** [comment]
Or I might be one of the very many FF users who don't use FF sync at all. :)

You should probably default that setting to off.
AND, when sync is enabled, don't sync older (or factory) on top of newer.

**Comment by SanderRonde** [comment]
I'm pretty sure it's not something you can turn on or off at all. It's enabled by default and simply syncs it to your other browser instances if here are any. If you're not logged in anywhere else, it acts just like a local storage so there's no difference for people who don't actually switch devices. 

I also don't really see a reason to disable it by default as there's really no downsides to it if this bug is fixed. The only thing I can think of is if someone wants different settings on different devices, in which case they can simply turn it off.

Regarding your second point, the new version that fixes this bug just went live (at least on firefox) so that shouldn't be a problem anymore.

**Comment by SanderRonde** [comment]
I think I've fixed the issue with the release (1862783913dc31a6b8c8169b8915d80250623995). Please let me know and re-open the issue if it happens again, but for now I'll be closing the issue.

**Comment by BrockA** [comment]
Thanks.

Meanwhile, I verified that [CCleaner](https://www.ccleaner.com/ccleaner) was trashing some add-on's data, unless it was carefully, hand, whitelisted.

After painfully (re)building new Firefox profiles I've switched to a different context-menu add-on for now.  It's closer to my needs and needs fewer permissions, But since it has a couple of annoyances and doesn't appear to be open-source, I may switch back to CRM.
